src/gstreamer/gstkvssink.cpp

```diff
@@ -1217,14 +1224,16 @@ bool put_frame(shared_ptr<KvsSinkCustomData> data, void *frame_data, size_t len,
     Frame frame;
     create_kinesis_video_frame(&frame, pts, dts, flags, frame_data, len, track_id, index);
     bool ret = data->kinesis_video_stream->putFrame(frame);
-    if(data->get_metrics && ret) {
-        if(CHECK_FRAME_FLAG_KEY_FRAME(flags)  || data->on_first_frame){
+        
+    if(CHECK_FRAME_FLAG_KEY_FRAME(flags)  || data->on_first_frame) {
+        data->fragment_metadata_count = data->persisted_fragment_metadata_count;
```

Comment:
Can you help me to understand why we don't set `fragment_metadata_count` to zero here? Seems like this would need to be reset to zero at the start of every fragment to not exceed the max tags per fragment.

And if `persisted_fragment_metadata_count` is a count of the total tags persisted throughout the whole stream, can we add a `total_` prefix to this variable name to make that more clear?

Reply:
Persisted tags by default means that we are going to add them to all the subsequent fragments so to respect the limit, it has been initialized to persisted_fragment_metadata_count. I did not include "total" in the name as "persisted", here, means that we are going to add it to every fragment from now on

Reply:
Ah got it. Can we consider renaming this to `persistent_` rather than `persisted_` in that case? And do you think we should also provide an upstream message type to remove a persistent tag?

Reply:
I don't think we have an API to remove the flag from persisted metadata actually. I named it "persisted" as they have been persisted into our media already in the previous calls

Reply:
As per: https://docs.aws.amazon.com/kinesisvideostreams/latest/dg/how-meta.html#how-meta-add

The way to "remove" the metadata is to call the API with an empty *value* will remove the metadata from the queue, we should support this logic as well.

Reply:
The customer can modify the function that sets the names and values of the metadata so I don't think this will need any modification. When Stefan suggested removing, I thought he was talking about a new KVS API to do that and we don't have that but I'm sure we can call the existing APIs with different values